src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2383,11 +2411,21 @@ private void PopulateGenericParameters(ImmutableArray<IGenericParameter> sortedG
 
                 foreach (var refWithAttributes in genericParameter.GetConstraints(Context))
                 {
-                    metadata.AddGenericParameterConstraint(
-                        genericParameter: genericParameterHandle,
-                        constraint: GetTypeHandle(refWithAttributes.TypeRef));
+                    if (builder == null)
+                    {
+                        builder = ArrayBuilder<HandleWithAttributes>.GetInstance();
+                    }
+
+                    builder.Add(new HandleWithAttributes(
```

Comment:
>builder.Add(new HandleWithAttributes [](start = 20, length = 36)

Doc-comment for MetadataBuilder.AddCustomAttribute says: "Entries may be added in any order. The table is automatically sorted when serialized." which means that the code can be simplified if we simply add attributes right here. The same simplification can be applied to implemented interfaces as well.